scripts/common.sh

```
@@ -1367,6 +1334,59 @@ verify_single_level_env_placeholders() {
+      if [[ "$_arr_line" == *'='* ]]; then
+        local _arr_key="${_arr_line%%=*}"
+        _arr_key="${_arr_key%%[[:space:]]*}"
+        [[ -n "$_arr_key" ]] && _arr_known_env["$_arr_key"]=1
+      fi
+    done <"$env_file"
+  fi
+
+  local _arr_unexpected=0
+  local _arr_placeholder="" _arr_name=""
+  local _arr_matches=""
+  _arr_matches="$(grep -o '\${[A-Za-z_][A-Za-z0-9_]*}' "$compose_file" 2>/dev/null | sort -u || true)"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)